infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -244,12 +331,22 @@ private KubernetesNamespaceMeta asNamespaceMeta(Namespace namespace) {
    * @throws InfrastructureException if any exception occurs during namespace preparing
    */
   public KubernetesNamespace create(String workspaceId) throws InfrastructureException {
-    final String namespaceName =
-        evalNamespaceName(workspaceId, EnvironmentContext.getCurrent().getSubject());
+    final String namespaceName;
+    try {
+      namespaceName = evalNamespaceName(workspaceId, EnvironmentContext.getCurrent().getSubject());
+    } catch (NotFoundException | ServerException | ConflictException | ValidationException e) {
+      throw new InfrastructureException(
+          format(
+              "Failed to determine the namespace to put the workspace %s to."
+                  + " The error message was: %s",
+              workspaceId, e.getMessage()),
+          e);
+    }
+
     KubernetesNamespace namespace = doCreateNamespace(workspaceId, namespaceName);
     namespace.prepare();
 
-    if (!isPredefined() && !isNullOrEmpty(serviceAccountName)) {
+    if (isCreatingNamespace(workspaceId) && !isNullOrEmpty(serviceAccountName)) {
```

Comment:
No. Because we (are attempting to) create `<username>` namespaces but we're not deleting them, unlike for `<workspaceid>` namespaces, which we manage fully.